torch/distributed/_tensor/api.py

```diff
@@ -343,6 +343,14 @@ def from_local(
         # set default placements to replicated if not specified
         if placements is None:
             placements = [Replicate() for _ in range(device_mesh.ndim)]
+        else:
+            placements = list(placements)
+            for idx, placement in enumerate(placements):
+                # normalize shard dim to be positive
+                if placement.is_shard():
+                    placement = cast(Shard, placement)
+                    if placement.dim < 0:
+                        placements[idx] = Shard(placement.dim + local_tensor.ndim)
 
```

Comment:
The conversion of `placements` to `tuple` is below:
https://github.com/pytorch/pytorch/blob/43a9a1fe209e245f2b45c23077f3f74be63e83d5/torch/distributed/_tensor/api.py#L358-L365